### PR TITLE
New methods SendFromCustomInputs and UnspentOutputs

### PR DESCRIPTION
**Pull request by Kifen**
Added two functions "SendFromCustomInputs" and "UnspentOutputs" - these functions were adapted from similar functions on branch dcrlibwallet-wip

### CODE REVIEW

File: txauthor.go [hunk 1]
```diff
@@ -3,8 +3,10 @@ package dcrlibwallet
 import (
 	"bytes"
 	"context"
+	"fmt"
```
Comment (itswisdomagain) [inline txauthor.go]: Re-arrange imports

File: txauthor.go [hunk 2]
```diff
@@ -184,3 +186,136 @@ func (lw *LibWallet) PublishUnminedTransactions() error {
 	err = lw.wallet.PublishUnminedTransactions(ctx, netBackend)
 	return err
 }
+
+func (lw *LibWallet) SignAndPublishTransaction(serializedTx, privPass []byte) ([]byte, error) {
```
Comment (itswisdomagain) [inline txauthor.go]: This should go below `SendFromCustomInputs`

File: txhelper/outputs.go [hunk 3]
```diff
@@ -1,12 +1,28 @@
...
 import (
+	"fmt"
+
 	"github.com/decred/dcrd/dcrutil"
 	"github.com/decred/dcrd/txscript"
 	"github.com/decred/dcrd/wire"
 	"github.com/raedahgroup/dcrlibwallet/addresshelper"
 )
 
+func makeTxOutputs(destinations []TransactionDestination) (outputs []*wire.TxOut, totalSendAmount int64, err error) {
```
Comment (itswisdomagain) [inline txhelper/outputs.go]: Rename this to `MakeTxOutputs`, remove the other `MakeTxOutputs` below.

File: txauthor.go [hunk 4]
```diff
@@ -184,3 +186,136 @@ func (lw *LibWallet) PublishUnminedTransactions() error {
 	err = lw.wallet.PublishUnminedTransactions(ctx, netBackend)
 	return err
 }
+
+func (lw *LibWallet) SendFromCustomInputs(sourceAccount uint32, requiredConfirmations int32, utxoKeys []string,
+	txDestinations []txhelper.TransactionDestination, changeDestinations []txhelper.TransactionDestination, privPass []byte) (string, error) {
+
+	// fetch all utxos in account to extract details for the utxos selected by user
+	// use targetAmount = 0 to fetch ALL utxos in account
```
Comment (itswisdomagain) [inline txauthor.go]: This comment is no longer relevant here. Should be moved to before line 86 of `accounts.go`. Also, rename the `UnspentOutputs` function to `AllUnspentOutputs`

File: txauthor.go [hunk 5]
```diff
@@ -184,3 +186,136 @@ func (lw *LibWallet) PublishUnminedTransactions() error {
...
+		outpoint := wire.NewOutPoint(txHash, utxo.OutputIndex, int8(utxo.Tree))
+		input := wire.NewTxIn(outpoint, int64(utxo.Amount), nil)
+		inputs = append(inputs, input)
+		totalInputAmount += input.ValueIn
+
+		if len(inputs) == len(utxoKeys) {
+			break
+		}
+	}
+
```
Comment (itswisdomagain) [inline txauthor.go]: Add a check here to ensure that `len(inputs) == len(utxoKeys)`:
```
if len(inputs) != len(utxoKeys) {
     return "", fmt.Errorf("could not find matching inputs for all provided utxo keys")
}
```

**Comment by itswisdomagain** [comment]
This will benefit from https://github.com/decred/dcrwallet/pull/1573

**Comment by RogueElement** [comment]
Is this still active?

**Comment by C-ollins** [comment]
This would be revisited later.

**Comment by C-ollins** [comment]
Backed up here https://github.com/raedahgroup/dcrlibwallet/tree/SendFromCustomInputs

**Pull request closed without merging**